gtda/diagrams/features.py

```diff
@@ -689,7 +689,7 @@ def transform(self, X, y=None):
 
         Xt = Parallel(n_jobs=self.n_jobs)(
             delayed(self._complex_polynomial)(
-                _subdiagrams(Xt[s], [dim], remove_dim=True),
+                _subdiagrams(Xt, [dim], remove_dim=True)[s],
```

Comment:
This is a bit absurd from the point of view of performance, at each iteration we are computing all subdiagrams and also allocating memory to all, when we only need one. It's true my change was wrong but I suggest this
```suggestion
                _subdiagrams(Xt[[s]], [dim], remove_dim=True),
```

Reply:
Although my general preference would be towards vectorizing this implementation, intuitively I thought it would be possible, I'm looking into it now.

Reply:
I believe that you are looking at an outdated version - the current version is
```
                _subdiagrams(Xt[[s]], [dim], remove_dim=True)[0],
```

I agree that it's not vectorized though.

Reply:
Oh ok, GitHub decided I needed an outdated diff. Thanks!

Reply:
I am not sure how to vectorize "roots -> coefficients of the polynomial" step - not with `np.poly` that we currently use.